x/supplier/types/errors.go

```diff
@@ -16,4 +16,6 @@ var (
 	ErrSupplierInvalidSessionId          = sdkerrors.Register(ModuleName, 1107, "invalid session ID")
 	ErrSupplierInvalidService            = sdkerrors.Register(ModuleName, 1108, "invalid service in supplier")
 	ErrSupplierInvalidSessionEndHeight   = sdkerrors.Register(ModuleName, 1109, "invalid session ending height")
+	ErrSupplierUnbonding                 = sdkerrors.Register(ModuleName, 1110, "supplier is in unbonding period")
```

Comment:
Are we sure about "unbonding" as opposed to "unstaking"? My understanding was that the "bond" nomenclature was an extension of the chemistry analogy of the "atom" token in the cosmos-hub network. I think here, it is less precisely defined.

Reply:
Are you suggesting to fully replace the `Unbonding` term by `Unstaking`?

Reply:
I suppose so. :sweat_smile:

Reply:
@Olshansk, any opinion on this?

Reply:
https://docs.cosmos.network/v0.47/build/modules/staking

The way I see it is that `Stake` / `Unstake` is an action (verb) triggered by an end user.

I realize that `Bond` / `Unbond` can also be a verb, but I see it as more of a status. See [cosmos/staking/v1beta1/staking.proto](https://github.com/cosmos/cosmos-sdk/blob/main/x/staking/proto/cosmos/staking/v1beta1/staking.proto)

![Screenshot 2024-07-15 at 1 35 11 PM](https://github.com/user-attachments/assets/69386f97-2010-49a2-a525-a231990b662c)

Reply:
Another thing worth considering is that we're not even using the terms "stake" and "staking" in the same sense as conventional cosmos-sdk conceives of it as in their picture, it's part of DPoS, which is not the case for us. I would argue that this accumulates in favor of avoiding these ambiguous terms in our unique context, where we've already appropriated and defined "stake" and "staking".

Reply:
@bryanchriswhite

1. In cosmos, the `x/saking` module is strictly for validators so I was using it as a reference.
2. I hear your argument and don't feel ultra strongly about it, but am still not convinced after doing some research.
3. See Ethereum & Solana docs for reference below where "staked assets" have "unbonding periods" after an "unstaking event".

Lmk if you still feel like we should rename it.

--- 

Source 1 (ethereum): https://support.bitpanda.com/hc/en-us/articles/8324987568284-Ethereum-Staking

<img width="740" alt="Screenshot 2024-07-16 at 8 14 12 PM" src="https://github.com/user-attachments/assets/c1f901f2-4817-4470-a822-cef436895031">

Source 2 (solana): https://docs.blockdaemon.com/docs/solana-specific-info

<img width="840" alt="Screenshot 2024-07-16 at 8 14 41 PM" src="https://github.com/user-attachments/assets/3e7b65af-152b-48eb-86ce-e21deccb8fce">

Reply:
@red-0ne I talked to Bryan about this. Can you just update the docs next to the enum/types making it very very clear what (un)staking and (un)bonding is?